pm/recipient_test.go

```diff
@@ -224,8 +225,8 @@ func TestReceiveTicket_ValidNonWinningTicket(t *testing.T) {
 	recipientRand := genRecipientRand(sender, secret, params)
 	senderNonce := r.(*recipient).senderNonces[recipientRand.String()]
 
-	if senderNonce.nonce != newSenderNonce {
-		t.Errorf("expected senderNonce to be %d, got %d", newSenderNonce, senderNonce.nonce)
+	if senderNonce.Front().Key != newSenderNonce {
```

Comment:
This is probably be me misunderstanding, but I'd expect this be `.Back()` to look at the most recent one

Reply:
Right, `Back()` would be most recent nonce, but in this test we test only one ticket anyway.